### ISSUE TITLE
'helm delete' reports spurious failure on 'delete' when a chart installed invalid yaml

### ISSUE THREAD

**Issue by marun** [kubernetes/helm#1493]
I tried to install a chart that had a stale temp file in the template path, which resulted in the following error:

```
Error: release flippant-seast failed: unable to decode "": no kind "Pod" is registered for version "extensions/v1beta1"
```

The release was correctly marked as ``FAILED``.  When subsequently invoking ``helm delete`` on the failed release, its associated resources were removed but the command returned ``1`` and a similar message as installation:

```
Error: deletion error count 1: unable to decode "": no kind "Pod" is registered for version "extensions/v1beta1"
```

I think it would make for a better experience to have deletion return successfully and without error unless something occurs that would prevent deletion of a release's resources.

cc: @technosophos

**Comment by technosophos** [comment]
I'm adding this to the 2.0.0 milestone for now. We may have to pull it until after 2.0.0, though.

**Comment by technosophos** [comment]
I've changed the error messages to be clearer, and changed `helm delete` to put this into the `DELETED` state. But it still returns 1 (error). I can't change that until after 2.0.0, and we're split over whether or not we should return 0 when in fact the deletion did not complete successfully.

**Comment by technosophos** [comment]
Closed by #1512
